cra-rxjs-styled-components/src/components/sub-header/SubHeader.tsx

```diff
@@ -29,32 +28,95 @@ import {
   PrIcon,
   IssuesIcon,
 } from '../icons/index';
+import { useParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { tap, forkJoin } from 'rxjs';
+import {
+  SINGLE_USER_REPO,
+  ISSUE_PR_SEARCH,
+} from '../../constants/url.constants';
+import { Repository } from '../../interfaces/repositories.interfaces';
+import { fromFetchWithAuth } from '../../hooks/auth/from-fetch-with-auth';
+
+type IssueDetails = {
+  total_count: number;
+  incomplete_results: boolean;
+  items: any[];
+};
 
-type Props = {
-  user: string;
-  repo: string;
-  privacy: boolean;
-  watchCount: number;
-  starCount: number;
-  forkCount: number;
-  issuesCount: number;
-  prCount: number;
+type RepositoryDetails = {
+  repo: Repository | null;
+  prs?: IssueDetails;
+  rootFileInfo?: any[];
+  issues?: IssueDetails;
 };
 
-export default function SubHeader({
-  user,
-  repo,
-  privacy,
-  watchCount,
-  starCount,
-  forkCount,
-  issuesCount,
-  prCount,
-}: Props) {
+export default function SubHeader() {
+  const [repoDetails, setRepoDetails] = useState<RepositoryDetails>({
+    repo: null,
+  });
+  const params = useParams();
+  const { repo, prs, issues } = repoDetails;
+  const request = (url: string) =>
+    fromFetchWithAuth(url, {
+      selector: (response: Response) => {
+        return response.json();
+      },
+    });
+
+  useEffect(() => {
+    forkJoin([
+      request(SINGLE_USER_REPO(params.username!, params.repo!)),
```

Comment:
It looks like most of these implementations will be repeated again in the individual `repository-code`, `repository-issues` and `repository-pull-request`.

Reply:
Also, I am sure this data is fetched here and is just used to get issues and PR counts?

Reply:
I don't like the non-null assertion operator in TypeScript. It's usually hiding a code smell.

Reply:
I used this branch as a parent for pr #339

Reply:
> Also, I am sure this data is fetched here and is just used to get issues and PR counts?

The data used is for subheader data like issues count and pr count.  The data from the issues had wrong counts as github counts PRs as issues.